src/addagent/CMakeLists.txt

```diff
@@ -0,0 +1,116 @@
+cmake_minimum_required(VERSION 3.12.4)
```

Comment:
this refactoring exceeds the scope of the original requirement. Only new tests should be added

Reply:
As I mentioned above I don't think is a bad idea to create a new CMakeList.txt, also, considering that we are not using Cmocka here. 

I'm thinking in something like this

![image](https://user-images.githubusercontent.com/13010397/214395610-46e96267-a608-4c46-ba09-381b672b039e.png)